Finish converting `o` commands to rzshell

ret2libc opened the pull request:
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [x] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed). See https://github.com/rizinorg/book/pull/88

**Detailed description**

<!-- Explain the **details** for making this change. Is a new feature implemented? What existing problem does the pull request solve? How does the pull request solve these issues? Please provide enough information so that others can review your pull request. -->

* Convert `oo` commands
* Convert `on` commands
* Remove `oi` commands
* Convert `oC` and `oc` commands
* Convert `o` and `o+` commands
* `o` without arguments becomes `ol` and `o.` becomes `ol.`

**Test plan**

<!-- What steps should the reviewer take to test your pull request? Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots/videos. This is your time to re-check that everything works and that you covered all the edge cases -->

...

**Closing issues**

<!-- put "closes #XXXX" in your comment to auto-close the issue that your PR fixes (if any). -->

#1589

ret2libc committed:
Move 'o' to 'ol' for listing opened files

ret2libc committed:
Move 'ox' to rzshell

ret2libc committed:
Convert 'o' and 'o+' to rzshell

ret2libc committed:
Convert 'oc' command to rzshell

ret2libc committed:
Convert 'oC' command to rzshell

ret2libc committed:
Remove 'oi' commands from rzshell

Relying on the index does not seem that useful.

ret2libc committed:
Convert 'on' commands to rzshell

ret2libc committed:
Convert 'oo' and all 'o' commands to rzshell

ret2libc committed:
Fix 'on' command

ret2libc committed:
Fix 'o' and 'o+' commands arguments

XVilka commented on librz/core/cmd/cmd_open.c:
```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+	if (addr == 0) { // if no baddr defined, use the one provided by the file
+		addr = UT64_MAX;
+	}
+	rz_core_bin_load(core, filepath, addr);
```
Please check the return value of this function

XVilka commented on librz/core/cmd/cmd_open.c:
```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+		}
+	}
+
+	rz_core_block_read(core);
```
Please handle the errors/check the return value

ret2libc replied:
this is never checked anywhere in the code base and I wouldn't know what to do in case of failure either. I don't think I should make the whole thing fail because of this.

wargio replied:
there is no reason to check that, because when it fails to read, the buffer is filled with 0xff as per design

XVilka commented on librz/core/cmd/cmd_open.c:
```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_io_desc_exchange(core->io, fd, fdx);
+	rz_core_block_read(core);
```
Please handle errors/check return

ret2libc replied:
Same above.

XVilka commented on librz/core/cmd/cmd_open.c:
```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+		goto err;
+	}
+
+	rz_core_bin_load(core, uri, 0);
```
Check the return value

XVilka commented on librz/core/cmd/cmd_open.c:
```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+	if (!data) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_io_read_at(core->io, core->offset, data, len);
```
Handle the errors here

XVilka commented on librz/core/cmd/cmd_open.c:
```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+	}
+
+	core->num->value = desc->fd;
+	rz_core_block_read(core);
```
Handle errors

ret2libc replied:
Same.

XVilka commented on librz/core/cmd/cmd_open.c:
```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+		}
+		fd = core->io->desc->fd;
+	}
+	rz_core_io_file_open(core, fd);
```
Check the return value please

ret2libc replied:
`RZ_IPI void rz_core_io_file_open(RzCore *core, int fd)`.

XVilka commented on librz/core/cmd/cmd_open.c:
```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+RZ_IPI RzCmdStatus rz_reopen_handler(RzCore *core, int argc, const char **argv) {
+	int fd;
+	if (argc > 1) {
+		fd = (int)rz_num_math(NULL, argv[1]);
```
Please check to prevent impossible `fd` values, e.g. negative ones

XVilka commented on librz/core/cmd/cmd_open.c:
```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+	int fd;
+	int perms = RZ_PERM_RW;
+	if (argc > 1) {
+		fd = (int)rz_num_math(NULL, argv[1]);
```
Same here

XVilka commented on librz/core/cmd/cmd_open.c:
```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+		fd = core->io->desc->fd;
+		perms |= core->io->desc->perm;
+	}
+	rz_core_io_file_reopen(core, fd, perms);
```
Check return

ret2libc replied:
`RZ_IPI void rz_core_io_file_reopen(RzCore *core, int fd, int perms)`

XVilka commented on librz/core/cmd/cmd_open.c:
```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+}
+
+RZ_IPI RzCmdStatus rz_reopen_binary_handler(RzCore *core, int argc, const char **argv) {
+	rz_core_file_reopen(core, argv[1], 0, 2);
```
Check return

XVilka commented on librz/core/cmd/cmd_open.c:
```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+		free(args[i]);
+	}
+	free(args);
+	rz_core_file_reopen_debug(core, args_str);
```
Check for return and handle the errors

ret2libc replied:
`RZ_API void rz_core_file_reopen_debug(RzCore *core, const char *args)`

XVilka commented on librz/core/cmd/cmd_open.c:
```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+RZ_IPI RzCmdStatus rz_reopen_debug_file_handler(RzCore *core, int argc, const char **argv) {
+	const char *uri = argv[1];
+	ut64 addr = argc > 2 ? rz_num_math(core->num, argv[2]) : 0;
+	rz_core_file_reopen_remote_debug(core, uri, addr);
```
Check return

ret2libc replied:
`RZ_API void rz_core_file_reopen_remote_debug(RzCore *core, const char *uri, ut64 addr)`

XVilka commented on librz/core/cmd/cmd_open.c:
```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+	rz_file_dump(file, (const ut8 *)s, strlen(s), 0);
+	rz_file_dump(file, (const ut8 *)"\n", 1, 1);
+	free(file);
+	rz_core_file_reopen_debug(core, "");
```
Check return

ret2libc replied:
`RZ_API void rz_core_file_reopen_debug(RzCore *core, const char *args)`

XVilka commented on librz/core/cmd/cmd_open.c:
```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+}
+
+RZ_IPI RzCmdStatus rz_reopen_nobin_handler(RzCore *core, int argc, const char **argv) {
+	rz_core_file_reopen(core, NULL, 0, 0);
```
Check return, etc

XVilka commented on librz/core/cmd/cmd_open.c:
```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+	if (!fname) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_core_bin_load_structs(core, fname);
```
Check return values and handle errors in those functions please

ret2libc replied:
Printed a warning, but I did not fail hard on this.

XVilka reviewed (changes requested):
Overall is good but needs better errors handling.

ret2libc committed:
Fix xvilka's review

ret2libc committed:
Missed bool2status for one usage

ret2libc committed:
Fix tests to use 'ol' instead of 'o'

XVilka reviewed (approved):
Looks ok now, but `ood` fails it seems in 4 tests, please fix them first.

XVilka commented on librz/core/cmd_descs/cmd_open.yaml:
```diff
@@ -13,6 +57,118 @@ commands:
+            cname: reopen_debug
+            summary: Reopen current file in debug mode
+            args:
+              - name: args
```
Missing `optional`

ret2libc committed:
Make <args> of ood command optional

ret2libc commented:
Before merging let's wait for me to finish the checklist:
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why. ==== There are few core_reopen API that I've used/touched but at the moment I do not have a clear picture of how they work of if they are all needed.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [x] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed) ====== See https://github.com/rizinorg/book/pull/88

ret2libc committed:
Merge branch 'dev' into o-rzshell

ret2libc committed:
Add some tests for 'o' commands
